packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform.dart

```diff
@@ -0,0 +1,327 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:in_app_purchase_android/in_app_purchase_android.dart';
+import 'package:in_app_purchase_android/src/types/in_app_purchase_exception.dart';
+import 'package:in_app_purchase_platform_interface/in_app_purchase_platform_interface.dart';
+
+import '../billing_client_wrappers.dart';
+
+/// [IAPError.code] code for failed purchases.
+const String kPurchaseErrorCode = 'purchase_error';
+
+/// [IAPError.code] code used when a consuming a purchased item fails.
+const String kConsumptionFailedErrorCode = 'consume_purchase_failed';
+
+/// [IAPError.code] code used when a query for previouys transaction has failed.
+const String kRestoredPurchaseErrorCode = 'restore_transactions_failed';
+
+/// Indicates store front is Google Play
+const String kIAPSource = 'google_play';
+
+/// An [InAppPurchasePlatform] that wraps Android BillingClient.
+///
+/// This translates various `BillingClient` calls and responses into the
+/// generic plugin API.
+class InAppPurchaseAndroidPlatform extends InAppPurchasePlatform
+    with WidgetsBindingObserver {
+  InAppPurchaseAndroidPlatform._()
+      : billingClient =
+            BillingClient((PurchasesResultWrapper resultWrapper) async {
+          _purchaseUpdatedController
+              .add(await _getPurchaseDetailsFromResult(resultWrapper));
+        }) {
+    if (InAppPurchaseAndroidPlatform.enablePendingPurchase) {
+      billingClient.enablePendingPurchases();
+    }
+
+    _readyFuture = _connect();
+    WidgetsBinding.instance!.addObserver(this);
+    _purchaseUpdatedController = StreamController.broadcast();
+    ;
+  }
+
+  /// Returns the singleton instance of the [InAppPurchaseAndroidPlatform].
+  static InAppPurchaseAndroidPlatform get instance => _getOrCreateInstance();
+  static InAppPurchaseAndroidPlatform? _instance;
+
+  /// Whether pending purchase is enabled.
+  ///
+  /// See also [enablePendingPurchases] for more on pending purchases.
+  static bool get enablePendingPurchase => _enablePendingPurchase;
+  static bool _enablePendingPurchase = false;
+
+  /// Enable the [InAppPurchaseConnection] to handle pending purchases.
+  ///
+  /// This method is required to be called when initialize the application.
+  /// It is to acknowledge your application has been updated to support pending purchases.
+  /// See [Support pending transactions](https://developer.android.com/google/play/billing/billing_library_overview#pending)
+  /// for more details.
+  /// Failure to call this method before access [instance] will throw an exception.
+  static void enablePendingPurchases() {
```

Comment:
This should also be in the addition class.

Reply:
Done.